### ISSUE TITLE
 Full expanded UI option

### ISSUE THREAD

**Issue by orcutt989** [gemrb/gemrb#2072]
#### Description
Add ability to expand the UI like the TUTU UI mod
![image](https://github.com/gemrb/gemrb/assets/5608027/1ad24d81-3f88-4556-933f-705af900e770)

**Comment by lynxlynxlynx** [comment]
Why not use that mod? If you don't have TUTU, install the widescreen mod instead. Our own scaling ideas are already tracked in #1315.

**Comment by orcutt989** [comment]
@lynxlynxlynx I dont want to use TUTU I want a vaninlla BG1 experience, hence the reason I'm using gemRB.

Installing the widescreen mod without a direct draw plugin or a GUI mod breaks the whole mod.  The whole reason I am using GemRB is to not have to use a ddraw plugin or GUI mod.
![image](https://github.com/gemrb/gemrb/assets/5608027/d9cc70b0-6790-4a6e-8bdb-ea350015ffb8)
![image](https://github.com/gemrb/gemrb/assets/5608027/09106cb1-1af7-469c-b433-cd7926c81a62)

**Comment by lynxlynxlynx** [comment]
We don't care about dlls shipped, we just use the modded gui files. Here's a slightly longer explanation if you haven't seen it yet:
https://gemrb.org/Features.html#wide-screen-higher-resolutions

If you're fine with the centered look, but want a bigger gui at 1920x1080, set the gemrb resolution to 960x540 and maximize the window.

**Comment by orcutt989** [comment]
Thanks I watched the video, but you suggest widescreen mod at the end, but from my screenshots I don't think using the mod works. What would you suggest for a "scaled" UI with gemRB without TUTU?

**Comment by lynxlynxlynx** [comment]
Without the mod it will always be centered and you have to resize the window, which will in turn resize everything drawn.
bg1 only supported 640x480, so it doesn't have files for anything bigger. bg2 does and that's what's in that screenshot.

re widescreen mod: make sure you're using the latest version. And yeah, it doesn't affect the main menu, just the ingame stuff.

**Comment by orcutt989** [comment]
Good to know thanks @lynxlynxlynx with regards to the resolution.

I'm not understanding how the widescreen mod will work.  I have the latest version installed. https://github.com/Gibberlings3/widescreen/releases/tag/v3.08 in the screenshots above where the UI is completely gone.

I noticed there is a `Removed obsolete GemRB mode (lynx)` (hehe) could that be related to why perhaps?

**Comment by lynxlynxlynx** [comment]
Those are fine, it's python errors that you should be looking for. Runtime exceptions and the like.

**Comment by orcutt989** [comment]
@lynxlynxlynx Ok so it seems like I'm not seeing any relevant errors in the console with the widescreen mod installed and the UI is missing. Any suggestions?

**Comment by lynxlynxlynx** [comment]
What happens if you press i for inventory?

**Comment by orcutt989** [comment]
> What happens if you press i for inventory?

![image](https://github.com/gemrb/gemrb/assets/5608027/935b0b97-f156-4964-94b2-d3984d98b5bd)

**Comment by lynxlynxlynx** [comment]
Hmm, looks like the mod did something wrong. The window is not centered and the black bar below is from it being too big. It's resized to the full height, so in effect the centering is not visible.

Looking at my last changes in 79415104fa, are your guiw08.chu and guiw10.chu in the game override folder identical? If not, you can try with the code changed to use guiw08 if that changes anything.

**Comment by lynxlynxlynx** [comment]
Huh?! Are any *.chu in override?

**Comment by orcutt989** [comment]
> Huh?! Are any *.chu in override?

Yes
![image](https://github.com/gemrb/gemrb/assets/5608027/f565d893-e185-45da-b7e7-42192a7443e4)

**Comment by lynxlynxlynx** [comment]
then try with copying guiw.chu to guiw10.chu

**Comment by orcutt989** [comment]
> then try with copying guiw.chu to guiw10.chu

Should I do this with before or after widescreen mod is installed?

**Comment by lynxlynxlynx** [comment]
You have it installed already, right? Everything points that way. Ok, so that's clearly better. Don't know if it dealt with the bottom bar better as well. I'll open a new issue to investigate, since that code snippet definitely needs more logic added for widescreen.

**Comment by orcutt989** [comment]
> You have it installed already, right? Everything points that way. Ok, so that's clearly better. Don't know if it dealt with the bottom bar better as well. I'll open a new issue to investigate, since that code snippet definitely needs more logic added for widescreen.

Awesome thanks!

**Comment by lynxlynxlynx** [comment]
Got a fix ready, but it's equivalent to what you've already done. The mod just doesn't pad the bottom window and leaves gaps.

**Comment by orcutt989** [comment]
So you mean "equivalent to what I've already done" as in the last screenshot I posted? Is there anyway we can scale the UI and center the dialog box?

**Comment by lynxlynxlynx** [comment]
Manually editing the CHU should be relatively easy (you could do that with NearInfinity). Or you could delete all the other new chus in override. But I don't really see the point — is all this hassle worthy for just having taller sidebars and less visible space?

**Comment by orcutt989** [comment]
"Not worth it" is a valid opinion i guess. 

It would just be nice to not have black and have the option of having floating UI gem-rb as-is functionality. Here is some screenshots of what it could look like.

![image](https://github.com/gemrb/gemrb/assets/5608027/fa9a0901-7c6a-4006-8cb9-958a68bb81fb)

![image](https://github.com/gemrb/gemrb/assets/5608027/080ca889-c4de-4e12-a114-9c6740d10852)

**Comment by lynxlynxlynx** [comment]
Ok, but that's different from what you were asking in the previous post. Both can be done with mods, just apparently not the base widescreen one. And unless it patched the exe to do this programmatically, it would work with gemrb as well.

On our side is just #1315. Replicating screenshot 1 would be silly, why give away the real estate? The mod used imagery padding, since the way the window is drawn in the game, even if those areas were transparent, you'd still see only black.

**Comment by orcutt989** [comment]
Sorry I'm not understanding what you mean "different".

The goal of this is to have a true "vanilla" experience without any other mods. This is almost the case with gemRB, but with the floating UI. You suggested using TUTU, but that is not my aim here. It would be nice if you could have a non-floating UI with gemRB as it seems to be the best way to have a vanilla experience without modding.

Then you suggested using the widescreen mod which requires other UI mods at higher resolutions.

My original description for this issue was one of the screenshots, I just posted a differenet one as a suggestion if that made it easier.

If you don't think its "worth it" I can understand, but I just thought it would be cool to have an option for a non-floating UI. I do think the floating UI is very cool, but some might like the full UI experience that is not feasible with mods or any other manner right now (obscure Polish UI scaling mod only works at 720p, TUTU is too heavy for vanilla experience).

**Comment by lynxlynxlynx** [comment]
I suggested using the tutu gui mod, since you posted it as an example. It works there, because bg2 is used to play bg1, meaning the base resources are available. For us the tutu requirement doesn't really matter, since gemrb can run all ~7 major engine versions. It's the modified/new data that counts (and can be perhaps taken from the mod without even installing it). The issue I closed this as a duplicate of will eventually enable a crude approximation without extra data, but that's it.
